tigramite/toymodels/context_model.py

```diff
@@ -0,0 +1,221 @@
+    (i.e. perform dummy projection)
+
+    links : dictionary
+        Ground truth links
+    node_classification : dictionary
+        Corresponds to ground truth links
+    """
+
+    # remove latent links, shift remaining, add dummy
+    augmented_links = {}
+    for node in node_classification.keys():
+        if node_classification[node] == "system":
+            keep_parents = []
+            for parent in links[node]:
+                if node_classification[parent[0][0]] == "system":
+                    keep_parents.append(parent)
+                elif node_classification[parent[0][0]] == "time_context":
+                    if parent[0][0] in observed_context_indices:
+                        keep_parents.append(((parent[0][0] - nb_latent_before(parent[0][0], observed_context_indices,
+                                                                              node_classification),
+                                              parent[0][1]), parent[1], parent[2]))
+                    else:
+                        keep_parents.append(((time_dummy_index, 0), 1., "dummy"))
+                elif node_classification[parent[0][0]] == "space_context":
+                    if parent[0][0] in observed_context_indices:
+                        keep_parents.append(((parent[0][0] - nb_latent_before(parent[0][0], observed_context_indices,
+                                                                              node_classification), parent[0][1]),
+                                             parent[1], parent[2]))
+                    else:
+                        keep_parents.append(((space_dummy_index, 0), 1., "dummy"))
+                augmented_links[node] = list(dict.fromkeys(keep_parents))
+
+        # remove all parents of context nodes
+        elif node_classification[node] == "time_context":
+            if node in observed_context_indices:
+                augmented_links[node - nb_latent_before(node, observed_context_indices, node_classification)] = []
+        elif node_classification[node] == "space_context":
+            if node in observed_context_indices:
+                augmented_links[node - nb_latent_before(node, observed_context_indices, node_classification)] = []
+
+    augmented_links[time_dummy_index] = []
+    augmented_links[space_dummy_index] = []
+    return augmented_links
+
+def shift_link_entries(links, const):
+    """
+    Helper Function to shift keys and values of a link dictionary by an integer constant.
+    """
+    shifted_links = {}
+    for key in links.keys():
+        shifted_key = key + const
+        values = links[key]
+        shifted_values = [((item + const, lag), c, f) for ((item, lag), c, f) in values]
+        shifted_links[shifted_key] = shifted_values
+    return shifted_links
+
+def group_links(links, node_types, node_type):
+    return {i: links[i] for i in links.keys() if node_types[i] == node_type}
+
+
+class ContextModel:
```

Comment:
Add transient_fraction because this is a parameter used in structural_causal_processes?